### PR TITLE
test(components): update a11y tests to use renderWithinLandmark

### LINKED ISSUE #427: a11y tests - create utility for tests to minimize code duplication
## Proposal

From @diego-codes  --

https://github.com/carbon-design-system/ibm-security/pull/375#discussion_r389133806
> Just a suggestion, it might be useful to create a [custom renderer](https://testing-library.com/docs/react-testing-library/setup#custom-render) for a11y tests that handle the creation of the main element and setting it as a container.

https://github.com/carbon-design-system/ibm-security/pull/375#discussion_r389135388
> Do all of Axe / DAP tests have the same expect statements? What if this whole test function is converted into a utility that reduces the code duplication? Something like: `testA11y(<FilterPanelAccordion title="test accordion title" />);`

### Acceptance criteria

- [ ] Custom renderer that handles the creation of the main element and setting it as a container
- [ ] Utility test that reduces code overhead for axe and DAP test declarations
- [ ] Existing accessibility tests are refactored to use proposed implementation

### PR DESCRIPTION
**Pull request by jendowns**
## Affected issues

- Resolves #427 

This PR includes a helper for rendering a component in a landmark element. 

However, in the interest of keeping tests readable & avoiding abstracted test cases, the two a11y assertions are kept as-is.

## Proposed changes

- add `renderWithinLandmark` helper to wrap jsx in a `main` landmark element
- update component a11y tests to use `renderWithinLandmark`
- update testing documentation

### CODE REVIEW

File: config/jest/helpers/renderWithinLandmark.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
```
Comment (SimonFinney) [inline config/jest/helpers/renderWithinLandmark.js:1]: 🔥

**Comment by SimonFinney** [comment]
Thanks @jendowns - this looks great, and I can definitely see it reducing our overhead when writing tests in future!

My only question - is it intentional that the utility test for reducing code overhead for axe and DAP test declarations (https://github.com/carbon-design-system/ibm-security/issues/427#issuecomment-595975056) is excluded from this PR?

**Comment by jendowns** [comment]
Hey @SimonFinney - yes, this is intentional. I don't want to generate tests in another file & abstract test cases and assertions. I prefer test cases that are very clear, and I don't agree that patterns/groupings of test assertions, such as with these two (2) lines of a11y assertions, are code duplication.

**Comment by jendowns** [comment]
I updated the PR description with more details 👍

**Pull request merged**